lib/extract-js.js

```diff
@@ -0,0 +1,231 @@
+var htmlparser = require('htmlparser2');
+var Errors = require('./errors');
+var rLineSplit = /\r\n|\r|\n/;
+var rHasNonWhitespace = /\S/;
+
+/**
+ * Html file representation (needed for errors output).
+ *
+ * @name HtmlFile
+ * @param {Object} params
+ * @param {String} params.filename
+ * @param {String} params.source
+ */
+var HtmlFile = function(params) {
+    params = params || {};
+    this._filename = params.filename;
+    this._lines = params.source.split(rLineSplit);
+};
+
+HtmlFile.prototype = {
+    /**
+     * Returns source filename for this object representation.
+     *
+     * @returns {String}
+     */
+    getFilename: function() {
+        return this._filename;
+    },
+
+    /**
+     * Returns array of source lines for the file.
+     *
+     * @returns {String[]}
+     */
+    getLines: function() {
+        return this._lines;
+    }
+};
+
+/**
+ * Parse html and retrieve script sources.
+ *
+ * @param {String} html
+ * @returns {Object[]}
+ */
+function getScripts(html) {
+    function onopen(name, attrs) {
+        // tag should be a <script>
+        if (name !== 'script' ||
+            // ignore scripts with src attribute
+            attrs.src ||
+            // script tag should has no type attribute or attribute should be equal to text/javascript
+            (attrs.type && attrs.type.toLowerCase() !== 'text/javascript')) {
```

Comment:
I suppose values other than `text/javascript` is really rare used. If you insist, I can add check for all of those values. But I thought it's good idea to start with `text/javascript` and add new values if needed depending on users feedback.